### PR TITLE
feat: Add support for not requiring SMTP auth with user management

### PR DESCRIPTION
**Pull request by xavier-calland**
- do not check for user/pass in isEmailSetUp
- do not pass an "auth" property to nodemailer if no user/pass

Issue: #3740

### CODE REVIEW

**Comment by nathanael-h** [comment]
Hello, on my server there is an SMTP which does not requiere auth when reached internally (via localhost or 127.0.0.1). I think this PR could fix the issue I have (https://github.com/YunoHost-Apps/n8n_ynh/issues/30). If I try to build and test this PR, would this help?

**Review by Joffcom — DISMISSED** [review]

**Comment by Joffcom** [comment]
Hey @xavier-calland,

Thanks for the PR and sorry for the delay, I am planning on getting this merged shortly.

File: packages/cli/src/UserManagement/email/NodeMailer.ts [hunk 1]
```diff
@@ -1,23 +1,32 @@
 /* eslint-disable @typescript-eslint/no-unsafe-assignment */
 import type { Transporter } from 'nodemailer';
 import { createTransport } from 'nodemailer';
+import type { IDataObject } from 'n8n-workflow';
```
Comment (netroy) [inline packages/cli/src/UserManagement/email/NodeMailer.ts:4]: please don't `IDataObject`. it's not much better than `any`.
`nodemailer` has the correct types for this. Please use `SMTPConnection.Options` or `SMTPTransport.Options`.

**Review by netroy — DISMISSED** [review]

File: packages/cli/src/UserManagement/email/NodeMailer.ts (hunk 1 above)
Comment (Joffcom) [reply packages/cli/src/UserManagement/email/NodeMailer.ts:4]: @netroy cheers, Will fix that in the morning.

**Review by Joffcom — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.3.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.3.0)

**Pull request merged**